packages/manager/modules/vps/src/migration/vps-migration.service.js

```diff
@@ -0,0 +1,96 @@
+import compact from 'lodash/compact';
+import find from 'lodash/find';
+import map from 'lodash/map';
+import set from 'lodash/set';
+
+import { EXCLUDED_VERSIONS } from './vps-migration.constants';
+
+export default class VpsMigrationService {
+  /* @ngInject */
+  constructor($q, $http, $translate, $timeout) {
+    this.$q = $q;
+    this.$http = $http;
+    this.$translate = $translate;
+    this.$timeout = $timeout;
+  }
+
+  getVpsList() {
+    return this.$http
+      .get(`/vps`)
+      .then(({ data }) =>
+        this.$q.all(map(data, (vps) => this.getVpsDetails(vps))),
```

Comment:
Could we use headers to get the expanded values and only have one call ?
 `X-Pagination-Mode`: `CachedObjectList-Pages`
(You can find details about the available headers : https://github.com/ovh-ux/ng-ovh-api-wrappers/blob/master/src/iceberg/request-upgrader.service.js#L24)